internal/cli/namespaces.go

```diff
@@ -101,31 +102,58 @@ var CmdNamespaceCreate = &cobra.Command{
 var CmdNamespaceDelete = &cobra.Command{
 	Use:               "delete NAME",
 	Short:             "Deletes an epinio-controlled namespace",
-	Args:              cobra.ExactArgs(1),
+	Args:              cobra.MinimumNArgs(0),
 	ValidArgsFunction: matchingNamespaceFinder,
-	PreRunE: func(cmd *cobra.Command, args []string) error {
-		force, err := cmd.Flags().GetBool("force")
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+
+		all, err := cmd.Flags().GetBool("all")
 		if err != nil {
-			return err
+			return errors.Wrap(err, "error reading option --all")
 		}
-		if !force {
-			cmd.Printf("You are about to delete namespace %s and everything it includes, i.e. applications, configurations, etc. Are you sure? (y/n): ", args[0])
-			if !askConfirmation(cmd) {
-				return errors.New("Cancelled by user")
-			}
+
+		if all && len(args) > 0 {
+			return errors.New("Conflict between --all and given namespaces")
+		}
+		if !all && len(args) == 0 {
+			return errors.New("No namespaces specified for deletion")
 		}
 
-		return nil
-	},
-	RunE: func(cmd *cobra.Command, args []string) error {
-		cmd.SilenceUsage = true
+		force, err := cmd.Flags().GetBool("force")
+		if err != nil {
+			return err
+		}
```

Comment:
No need to check and get the value of `force`.

Reply:
See above.

Reply:
Done, commit 529aea3402fe8b101ae32f79858d6e0a50e607e9